components/FormValidator.js

```diff
@@ -0,0 +1,84 @@
+export default class FormValidator {
+  constructor(
+    {
+      inputSelector, //"donations__input"
+      submitButtonSelector, //"donations__submit-button"
+      inactiveButtonClass, // "donations__submit-button_type_inactive"
+      inputErrorClass, // "donations__input_type_error"
+      errorClass, //"donations__input-error_type_active"
+    },
+    formElement //"donations__form"
+  ) {
+    this._element = formElement;
+    this._inputSelector = inputSelector;
+    this._submitButtonSelector = submitButtonSelector;
+    this._inactiveButtonClass = inactiveButtonClass;
+    this._inputErrorClass = inputErrorClass;
+    this._errorClass = errorClass;
+  }
+
+  //Show error
+  _showInputError(input, errorMessage) {
+    const errorElement = this._element.querySelector(`#${input.id}-error`);
+    input.classList.add(this._inputErrorClass);
+    errorElement.textContent = errorMessage;
+    errorElement.classList.add(this._errorClass);
+  }
+
+  //Hide error
+  _hideInputError(input) {
+    const errorElement = this._element.querySelector(`#${input.id}-error`);
+    input.classList.remove(this._inputErrorClass);
+    errorElement.classList.remove(this._errorClass);
+    errorElement.textContent = "";
+  }
+
+  //Check form field
+  _isValid(input) {
+    if (!input.validity.valid) {
+      this._showInputError(input, input.validationMessage);
+    } else {
+      this._hideInputError(input);
+    }
+  }
+
+  //Check all fields in form
+  _setEventListeners() {
+    const inputList = Array.from(
+      this._element.querySelectorAll(this._inputSelector)
+    );
+    const submitBtn = this._element.querySelector(this._submitButtonSelector);
+    this._toggleButtonState(inputList, submitBtn);
+
+    inputList.forEach((input) => {
+      this._isValid(input);
+      input.addEventListener("input", () => {
```

Comment:
General comment, you should make all your strings single quote to improve consistency.

Reply:
All my strings are consistent and double quote - that's the way my save&format plugin works

Reply:
Can you change it to single? We originally had single quote as our string type. We don't wanr some files made with single quote and other files with double quotes.

Reply:
We already have double quotes in index.html, by the way :)